File: go-for-pythonistas/udt.go
package main

import "fmt"

//An object in go is called a struct
type Person struct {
    Firstname string
    Lastname string
    Phone string
}

//you can attach methods to a struct
func (p *Person) Info() string {
    return fmt.Sprintf("%v %v (%v)", p.Firstname, p.Lastname, p.Phone)
}

func main() {
    p := Person{"Dan", "Worth", "800-555-5555"}
    fmt.Println("Hello", p.Info())
}

File: go-for-pythonistas/udt.py


class Person(object):

    def __init__(self, firstname, lastname, phone):
        self.firstname = firstname
        self.lastname = lastname
        self.phone = phone

    def info(self):
        return "{} {} ({})".format(self.firstname, self.lastname, self.phone)


def main():
    p = Person("Dan", "Worth", "800-555-5555")
    print "Hello {}".format(p.info())

if __name__ == "__main__":
    main()
